automatic/youtube-dl/update.ps1

```diff
@@ -1,20 +1,36 @@
 import-module au
+import-module "$env:ChocolateyInstall\helpers\chocolateyInstaller.psm1"
 
 $domain   = 'https://github.com'
 $releases = "$domain/rg3/youtube-dl/releases/latest"
 
-function global:au_afterUpdate {
-  rm $PSScriptRoot/tools/*.exe -ea Ignore
+function global:au_BeforeUpdate {
+  if (Test-Path "$PSScriptRoot\tools") {
+    Remove-Item "$PSScriptRoot\tools\*.exe" -Force
+  } else {
+    New-Item -ItemType Directory "$PSScriptRoot\tools"
+  }
+  $Latest.FileName = Get-WebFileName $Latest.URL32 'youtube-dl.exe'
+  $filePath = "$PSScriptRoot\tools\$($Latest.FileName)"
+  Get-WebFile $Latest.URL32 $filePath
+
+  # Let us check if the checksum actually matches
+  $actualChecksum = Get-FileHash -Algorithm $Latest.ChecksumType32 $filePath | % Hash
+
+  if ($actualChecksum -ne $Latest.Checksum32) {
```

Comment:
it isn't set by `Get-WebFile`
`Get-WebFile` is a choco function to just download the file (you're thinking of `Get-RemoteFiles`).
the checksum is set using the provided checksum by the software developers.
So lets say something happens while downloading the file, but an error isn't reported the generated checksum wont match the provided checksum.

The check is probably almost unnecessary, but I feel it's better to have 1 check to many than 1 to few